src/components/main/Main.tsx

```diff
@@ -1,26 +1,67 @@
 import styles from './Main.css';
 import commonStyle from '../../styles/commonStyles.css';
 
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { usePosition } from 'hooks/usePosition';
-
-interface MainProps {
-  positionData: {
-    latitude?: number;
-    longitude?: number;
-  };
-  positionError?: string | null;
-}
+import { getCity } from 'services/getCity';
+import { getCityWeather, cityWeatherType } from 'services/getWeather';
 
 const Main: FC = () => {
   const { position, error } = usePosition();
+  const [city, setCity] = useState<string>('');
+  const [weather, setWeather] = useState<cityWeatherType>({
+    current: {
+      time: '',
+      symbol: '',
+      symbolPhrase: '',
+      temperature: 0,
+      feelsLikeTemp: 0,
+      relHumidity: 0,
+      dewPoint: 0,
+      windSpeed: 0,
+      windDir: 0,
+      windDirString: '',
+      windGust: 0,
+      precipProb: 0,
+      precipRate: 0,
+      cloudiness: 0,
+      thunderProb: 0,
+      uvIndex: 0,
+      pressure: 0,
+      visibility: 0
+    }
+  });
+
+  async function fetchData(lon: number, lat: number) {
+    await getCity(lon, lat).then(locationData => setCity(locationData.city));
+    await getCityWeather(lon, lat).then(cityWeather => setWeather(cityWeather));
+  }
+
+  useEffect(() => {
+    if (position.longitude) {
+      const lon = position.longitude as number;
```

Comment:
`as` should be used only in exceptional cases when there is no way to define typings directly (when 3rd party libraries used, for example) so please refactor the code